core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -45,6 +45,56 @@ trait EntityDecoder[F[_], T] { self =>
       override def consumes: Set[MediaRange] = self.consumes
     }
 
+  def handleError(f: DecodeFailure => T)(implicit F: Functor[F]): EntityDecoder[F, T] =
+    new EntityDecoder[F, T] {
+      override def consumes: Set[MediaRange] = self.consumes
+
+      override def decode(msg: Message[F], strict: Boolean): DecodeResult[F, T] =
+        self.decode(msg, strict).recover(PartialFunction(f))
+    }
+
+  def handleErrorWith(f: DecodeFailure => DecodeResult[F, T])(
+      implicit F: Monad[F]): EntityDecoder[F, T] = new EntityDecoder[F, T] {
+    override def consumes: Set[MediaRange] = self.consumes
+
+    override def decode(msg: Message[F], strict: Boolean): DecodeResult[F, T] =
+      self.decode(msg, strict).leftFlatMap(f)
+  }
+
+  def transform[T2](s: T => T2, f: DecodeFailure => DecodeFailure)(
```

Comment:
Why the `DecodeFailure => DecodeFailure`? wouldn't you want this to be `DecodeFailure => T` or `Either[E, T]` ?

Reply:
This override is so you can have strict transformation of success (`T => T2`) and error cases (`DecodeFailure => DecodeFailure`) separately. 

Mainly inspired by `Future#transform` in the standard lib.

Reply:
hmm ok @lloydmeta, it just seems a bit inconsistent wrt [EitherT](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/data/EitherT.scala#L86), but this is only my opinion. If you feel this use case is quite common im not specifically against it.

:+1: thank you for the PR btw!

Reply:
@jmcardon I completely here you :) My hope is that the [other override for transform](https://github.com/http4s/http4s/pull/1575/files#diff-02708fea78424f8a8fa6eb4324a9daacR66) will make it feel more like that transform on `EitherT`.

Reply:
I saw that one as well, I just meant the having two methods part of the `transform`.

Either way, this is cool with me :+1: